src/main/kotlin/racingcar/domain/TryNumber.kt

```diff
@@ -0,0 +1,23 @@
+package racingcar.domain
+
+private const val MIN_TRY_NUMBER = 1
+private const val USE_COMPLETED_TRY_NUMBER = 0
+
+class TryNumber(value: Int) {
+    var value: Int = value
+        private set
+
+    init {
+        require(value >= MIN_TRY_NUMBER) { "시도횟수는 1이상인 수만 입력받을 수 있습니다. value = $value" }
+    }
+
+    fun consume() {
+        if (this.isComplete()) {
+            throw IllegalStateException("잔여 시도횟수가 없습니다.")
+        }
+
+        value--
```

Comment:
일괄적으로 `this`를 사용하는건 어떨까요?
```suggestion
        if (this.isComplete()) {
            throw IllegalStateException("잔여 시도횟수가 없습니다.")
        }

        this.value--
```

Reply:
에구구....`this` 썼다가 안썻다가... 상당히 일관성이 없는 코드였네요. 🙄